src/main/java/nitezh/ministock/activities/GlobalWidgetData.java

```diff
@@ -23,4 +40,187 @@
+        chds = "chds=a";                                            //Automatic text format scaling
+        chxt = "chxt=x%2Cy";                                         //Specify X and Y Axes
+        chs = "chs=700x690";                                        //chart size
+        chtt = "chtt=Test%20Graph";                                 //Name of Graph
+        String chdVars = "";
+
+        for (int i = 0; i < list.size(); i++) {
+            if (i == 0) {
+                chdVars = chdVars + list.get(i);                //first
+            } else
+                chdVars = chdVars + "%2C" + list.get(i);        //middle & end
+        }
+        String toReturn = "https://image-charts.com/chart?" + cht + "&" + chd + chdVars + "&" + chds + "&chof=.png&"
+                + chs + "&chdls=000000&chco=F56991%2CFF9F80%2CFFC48C%2CD1F2A5%2CEFFAB4&" + chtt + "&"
+                + chxt + "&chdlp=b&chf=bg%2Cs%2CFFFFFF&chbh=10&icwt=false";
+        return toReturn;
+    }
+}
+
+
+// Graph Implementation: Extracts values from a Json file
+class JsonSnatcher extends AsyncTask<String, Void, List<String>> {
+    String sURL;
+    int interval;
+
+    public JsonSnatcher() {
+    }
+
+    public JsonSnatcher(String sURL, int interval) {
+        this.sURL = sURL;
+        this.interval = interval;
+    }
+
+    @Override
+    protected List<String> doInBackground(String... strings) {
+        List<String> closingValuesWeekly = new ArrayList<>();
+
+        try {
+            URL url = new URL(strings[0]);
+            HttpURLConnection request = (HttpURLConnection) url.openConnection();
+            request.connect();
+
+            //Convert to a JSON object to print data
+            JsonParser jp = new JsonParser(); //use gson
+            JsonElement root = jp.parse(new InputStreamReader((InputStream)
+                    request.getContent()));
+
+            // Get a handle on the root
+            JsonObject rootObj = root.getAsJsonObject();
+
+            if (this.interval == 1) {
+                // Get a handle on Daily closing values
+                JsonObject dailyObj = rootObj.getAsJsonObject("Time Series (Daily)");
+
+                Set<Map.Entry<String, JsonElement>> entries = dailyObj.entrySet();
+                int dayCounter = 0;
+
+                for (Map.Entry<String, JsonElement> entry : entries) {
+                    if (dayCounter <= 6) {
+                        JsonObject dailyStats = entry.getValue().getAsJsonObject();
+                        closingValuesWeekly.add(dailyStats.getAsJsonPrimitive("4. close")
+                                .toString().replace("\"", ""));
+
+                        // Testing purposes
+                        Log.i("DayTest", dailyStats.getAsJsonPrimitive("4. close")
+                                .toString().replace("\"", ""));
+                    }
+                    dayCounter++;
+                }
+            }
+            if (this.interval == 3) {
```

Comment:
This should be an else-if or refactored to a switch for further flexibility, IMO

Reply:
fixed